src/hotspot/share/oops/fieldInfo.cpp

```diff
@@ -113,30 +121,222 @@ Array<u1>* FieldInfoStream::create_FieldInfoStream(GrowableArray<FieldInfo>* fie
   return fis;
 }
 
+int FieldInfoStream::compare_name_and_sig(const Symbol* n1, const Symbol* s1, const Symbol* n2, const Symbol* s2) {
+  int cmp = n1->fast_compare(n2);
+  return cmp != 0 ? cmp : s1->fast_compare(s2);
+}
+
+
+// We use both name and signature during the comparison; while JLS require unique
+// names for fields, JVMS requires only unique name + signature combination.
+typedef struct {
```

Comment:
Style: Use C++ struct def (no typedef), don't use trailing _t in the name and write it as FieldPos. The fields are public, so no underscore as name prefix for them.

Reply:
Now I am confused; @iklam just requested to use the underscores.

Reply:
This typedef didn't really bother me because it's a local internal class.  The coding style might be that it has to declared with camel case though.